src/postgres/tree/PostgresServerTreeItem.ts

```diff
@@ -68,16 +76,37 @@ export class PostgresServerTreeItem extends AzureParentTreeItem<ISubscriptionCon
     }
 
     public async loadMoreChildrenImpl(_clearCache: boolean): Promise<AzExtTreeItem[]> {
-        const client: PostgreSQLManagementClient = createAzureClient(this.root, PostgreSQLManagementClient);
-        const listOfDatabases: DatabaseListResult = await client.databases.listByServer(this.resourceGroup, this.name);
-        return this.createTreeItemsWithErrorHandling(
-            listOfDatabases,
-            'invalidPostgresServer',
-            (database) => {
-                return database.name && !['azure_maintenance', 'azure_sys'].includes(database.name) ? new PostgresDatabaseTreeItem(this, database.name) : undefined;
-            },
-            (database) => database.name
-        );
+        if (!(this.connectionString.username && this.connectionString.password)) {
+            const { username, password } = await this.getCredentials();
+            if (username && password) {
+                await this.updateConnectionString(username, password);
+            }
+        }
+        if (this._azureId) {
+            const client: PostgreSQLManagementClient = createAzureClient(this.root, PostgreSQLManagementClient);
+            const listOfDatabases: DatabaseListResult = await client.databases.listByServer(nonNullProp(this, 'resourceGroup'), nonNullProp(this, 'azureName'));
+            return this.createTreeItemsWithErrorHandling(
+                listOfDatabases,
+                'invalidPostgresServer',
+                (database) => {
+                    return database.name && !['azure_maintenance', 'azure_sys'].includes(database.name) ? new PostgresDatabaseTreeItem(this, database.name) : undefined;
+                },
+                (database) => database.name
+            );
+        } else {
+            const config = await getClientConfig(this, postgresDefaultDatabase);
+            const query = `SELECT datname FROM pg_catalog.pg_database WHERE datistemplate = false;`;
+            const queryResult = await runPostgresQuery(config, query);
+            const listOfDatabases = queryResult.rows;
+            return this.createTreeItemsWithErrorHandling(
```

Comment:
Can you find a way to get rid of some of this duplicate code with `createTreeItemsWithErrorHandling `

Reply:
I tried doing that but it would just create too many extra variables, given listOfDatabases and individual databases are both different types for azure and non azure scenario.